common/roschannels/monitorchannel.hpp

```diff
@@ -22,12 +30,99 @@ namespace ROSChannels {
+                BaseSub<message_type>(node, "object_anchor/" + topicName, callback, qos) {}
+        };
+
+        static message_type fromISOMonr(const uint32_t id, const ObjectMonitorType& indata) {
+            atos_interfaces::msg::Monitor outdata;
+            auto txid = id;
+            auto stamp = rclcpp::Time(indata.timestamp.tv_sec, indata.timestamp.tv_usec*1000);
+            
+            // Set stamp for all subtypes
+            outdata.atos_header.header.stamp = stamp;
+            outdata.pose.header.stamp = stamp;
+            outdata.velocity.header.stamp = stamp;
+            outdata.acceleration.header.stamp = stamp;
+
+            // Set frame ids
+            outdata.atos_header.header.frame_id = "map"; // TODO
+            outdata.pose.header.frame_id = "map"; // TODO
+            outdata.velocity.header.frame_id = "map"; // TODO vehicle local
+            outdata.acceleration.header.frame_id = "map"; // TODO vehicle local
+
+            outdata.atos_header.object_id = txid;
+            outdata.object_state.state = indata.state;
+            if (indata.position.isPositionValid) {
+                outdata.pose.pose.position.x = indata.position.xCoord_m;
+                outdata.pose.pose.position.y = indata.position.yCoord_m;
+                outdata.pose.pose.position.z = indata.position.isZcoordValid ? indata.position.zCoord_m : 0.0;
+            }
+            if (indata.position.isHeadingValid) {
+                tf2::Quaternion orientation;
+                orientation.setRPY(0, 0, indata.position.heading_rad);
+                outdata.pose.pose.orientation = tf2::toMsg(orientation);
+            }
+            outdata.velocity.twist.linear.x = indata.speed.isLongitudinalValid ? indata.speed.longitudinal_m_s : 0;
+            outdata.velocity.twist.linear.y = indata.speed.isLateralValid ? indata.speed.lateral_m_s : 0;
+            outdata.velocity.twist.linear.z = 0;
+            outdata.velocity.twist.angular.x = 0;
+            outdata.velocity.twist.angular.y = 0;
+            outdata.velocity.twist.angular.z = 0;
+            outdata.acceleration.accel.linear.x = indata.acceleration.isLongitudinalValid ? indata.acceleration.longitudinal_m_s2 : 0;
+            outdata.acceleration.accel.linear.y = indata.acceleration.isLateralValid ? indata.acceleration.lateral_m_s2 : 0;
+            outdata.acceleration.accel.linear.z = 0;
+            outdata.acceleration.accel.angular.x = 0;
+            outdata.acceleration.accel.angular.y = 0;
+            outdata.acceleration.accel.angular.z = 0;
+            return outdata;
+        }
+
+        static ObjectMonitorType toISOMonr(message_type& indata){
+            ObjectMonitorType outdata;
+            outdata.timestamp.tv_sec = indata.atos_header.header.stamp.sec;
+            outdata.timestamp.tv_usec = indata.atos_header.header.stamp.nanosec / 1000;
+            //TODO: Add state..
```

Comment:
Will state be added in an additional PR? There is a OBJECT_STATE_UNKNOWN that could be used instead of leaving garbage data in the state slot before this is handled.

Reply:
I just forgot to do this, thanks!